extensions/wikia/EmbeddableDiscussions/models/DiscussionsThreadModel.class.php

```diff
@@ -20,27 +16,69 @@ public function __construct( $cityId ) {
 		$this->cityId = $cityId;
 	}
 
-	// TODO: Refactor this to use Swagger, see JPN-631.
-	private function getRequestUrl( $showLatest, $limit ) {
+	// TODO: Consider changing this request to use Swagger when unblocked. See JPN-631
+	private function apiRequest( $url ) {
+		return json_decode( Http::get( $url ), true );
+	}
+
+	private function getCategoryRequestUrl() {
+		global $wgDevelEnvironment;
+		if ( empty( $wgDevelEnvironment ) ) {
+			return self::DISCUSSIONS_API_BASE . "$this->cityId/forums?responseGroup=small&viewableOnly=true";
+		}
+		return self::DISCUSSIONS_API_BASE_DEV . "$this->cityId/forums?responseGroup=small&viewableOnly=true";
+	}
+
+	public function getCategoryId( $category ) {
+		$memcKey = wfMemcKey( __METHOD__, self::MCACHE_VER );
+		$rawData = WikiaDataAccess::cache(
+			$memcKey,
+			WikiaResponse::CACHE_STANDARD,
+			function() {
+				return $this->apiRequest( $this->getCategoryRequestUrl() );
+			}
+		);
+
+		return $this->categoryLookup( $category, $rawData );
+	}
+
+	private function categoryLookup( $category, $rawData ) {
+		$categories = $rawData['_embedded']['doc:forum'];
+		if ( is_array( $categories ) ) {
+			foreach ( $categories as $value ) {
+				if ( $value['name'] === $category ) {
+					return $value['id'];
+				}
+			}
+		}
+
+		return false;
+	}
+
+	private function getRequestUrl( $showLatest, $limit, $category ) {
 		global $wgDevelEnvironment;
 
 		$sortKey = $showLatest ? self::SORT_LATEST : self::SORT_TRENDING;
+		$categoryId = $this->getCategoryId( $category );
+		$categoryKey = $categoryId ? '&forumId=' . $categoryId : '';
 
 		if ( empty( $wgDevelEnvironment ) ) {
-			return self::DISCUSSIONS_API_BASE . "$this->cityId/threads?sortKey=$sortKey&limit=$limit&viewableOnly=false";
+			return self::DISCUSSIONS_API_BASE .
+				"$this->cityId/threads?sortKey=$sortKey&limit=$limit&viewableOnly=false" . $categoryKey;
 		}
 
-		return self::DISCUSSIONS_API_BASE_DEV . "$this->cityId/threads?sortKey=$sortKey&limit=$limit&viewableOnly=false";
+		return self::DISCUSSIONS_API_BASE_DEV .
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)